### ISSUE TITLE
on /token endpoint -> keyObject.asInput is not a function at sign

### ISSUE THREAD

**Issue by stefdaems** [panva/node-oidc-provider#528]
**Describe the bug**
Whenever I send the a code retrieved from the authorization flow towards the /token endpoint I do get server_error keyObject.asInput is not a function at sign.

**To Reproduce**
Provider and Client configuration: (inline or gist)
```js
// provider configuration
As in your examples

// client(s) configuration
{
    client_id: 'yyyy_app',
    client_secret: 'xxxxxx,
    grant_types: ['authorization_code'],
    response_types: ['code'],
    redirect_uris: ["https://openidconnect.net/callback"],
    token_endpoint_auth_method: 'client_secret_basic'
}
```

Steps to reproduce the behaviour:

1.  authorize correctly
2.  send token to /token endpoint
3.  

**Expected behaviour**
claims are retrieved
**Environment:**
 - oidc-provider version: v6.6.2
 - node version: v11.9.0

**Additional context**
Test executed via https://openidconnect.net playground

 - [x ] the bug is happening on latest oidc-provider too.
 - [x ] i have tried DEBUG (see README) and can see the issue is with the provider and not my code.
 - [x ] i have searched the issues tracker on github for similar issues and couldn't find anything related.
 - [ ] i have searched the [FAQ](https://github.com/panva/node-oidc-provider/wiki/FAQ) and couldn't find anything related.

**Logs**
2019-09-10T06:56:49.271Z oidc-provider:token accepted uid=2nK6LpTJRYPXjhCtZQAFJ Params { grant_type: 'authorization_code', code: 'uq_hJoAmhbfKW61G3VV7yTk59-rmb9lzx1QaOyFcVyG', redirect_uri: 'https://openidconnect.net/callback', code_verifier: undefined, client_id: 'xxx_app', client_assertion: undefined, client_assertion_type: undefined, client_secret: ‘yyyyyyy’ }
2019-09-10T06:56:49.272Z oidc-provider:uid switched from uid=2nK6LpTJRYPXjhCtZQAFJ to value of grantId=7_FGi3sf-aY80Tr39SMqt
{"X-Request-Id":"","app”:”xxxx-service","app_revision":"08a1d608b34b450b240d7be3498998937ae17d26","app_version":"1909.06.08-BUILD","app_branch":"develop","stack":"openid-service","level":"INFO","message":"findAccount called with id: [object Object] and token: [object Object]","@timestamp":"2019-09-10T06:56:49.272Z"}
findAccount called with id: [object Object] and token: [object Object]
2019-09-10T06:56:49.276Z oidc-provider:server_error uid=7_FGi3sf-aY80Tr39SMqt path=/token method=POST error=TypeError: keyObject.asInput is not a function at sign (/opt/xxxx_service/node_modules/@panva/jose/lib/jwa/rsassa.js:15:63) at sign (/opt/xxxx_service/node_modules/@panva/jose/lib/jwa/index.js:58:29) at Sign.[PROCESS_RECIPIENT] (/opt/xxxxx_service/node_modules/@panva/jose/lib/jws/sign.js:108:50) at Array.forEach (<anonymous>) at Sign.sign (/opt/xxxx_service/node_modules/@panva/jose/lib/jws/sign.js:126:24) at single (/opt/xxxxx_service/node_modules/@panva/jose/lib/jws/index.js:7:14) at Function.sign (/opt/xxxxx_service/node_modules/oidc-provider/lib/helpers/jwt.js:49:16) at AccessToken.getValueAndPayload (/opt/xxxx_service/node_modules/oidc-provider/lib/models/formats/jwt.js:88:27) at processTicksAndRejections (internal/process/task_queues.js:86:5)
2019-09-10T06:56:49.276Z oidc-provider:server_error:trace TypeError: keyObject.asInput is not a function
    at sign (/opt/xxxxx_service/node_modules/@panva/jose/lib/jwa/rsassa.js:15:63)
    at sign (/opt/xxxxx_service/node_modules/@panva/jose/lib/jwa/index.js:58:29)
    at Sign.[PROCESS_RECIPIENT] (/opt/xxxxx_service/node_modules/@panva/jose/lib/jws/sign.js:108:50)
    at Array.forEach (<anonymous>)
    at Sign.sign (/opt/xxxxx_service/node_modules/@panva/jose/lib/jws/sign.js:126:24)
    at single (/opt/xxxxx_service/node_modules/@panva/jose/lib/jws/index.js:7:14)
    at Function.sign (/opt/xxxxx_service/node_modules/oidc-provider/lib/helpers/jwt.js:49:16)
    at AccessToken.getValueAndPayload (/opt/xxxxx_service/node_modules/oidc-provider/lib/models/formats/jwt.js:88:27)
    at processTicksAndRejections (internal/process/task_queues.js:86:5)
Tue, 10 Sep 2019 06:56:49 GMT morgan log request

**Comment by panva** [comment]
Hi,

node version v11.9.0 is not a supported runtime. Can you retry with one that’s supported first?

**Comment by stefdaems** [comment]
Doesn't think the node version has got anything to do with it. The application is running in a docker container, just fyi. 
Tried it with other node version (12.10.0) and there it is working.

What can be causing the error to be thrown?

**Comment by panva** [comment]
It is the node version. 

1) v11 has reached its end of life on 2019-06-01
2) @panva/jose - the underlying JOSE library is backfilling missing features from node 10 that aren't available in node 12, to do that its taking into consideration only the differences between v10 and v12, v11 was in a state where something was present thats in node 12 but not all.

Please use the indicated supported runtime version.

**Comment by stefdaems** [comment]
Solved idd, changed node version to v12.10 and issue no longer exists.
Thanks for the quick feedback and help!
